### ISSUE TITLE
Stepping away from Caliburn.Micro, transition plans and discussion

### ISSUE THREAD

**Issue by nigel-sampson** [Caliburn-Micro/Caliburn.Micro#708]
Earlier today I [announced][post] that I would be stepping away from maintaining Caliburn.Micro. It was a hard decision, but when I was honest with myself I realized I didn't have the time, nor the motivation due to where I am in my life these days. It would be unfair to the people still using the framework or looking to start using it to pretend otherwise. 

So what does this mean for the project. I'll publish the current version in the coming days as a release candidate for people to use.

The question I have for you all is, does it make sense for me to publish this as a full `4.0` release, without adequate support and documentation I fear that actually releasing it out of a pre-release stage may actually add more confusion.

I'm also keen to start a discussion with anyone looking to take up maintaining Caliburn.Micro and what an orderly transition would look like. If you feel like you can make the commitment then let me know.

Thank you all for the help and support you've given me over the eight years I've looked after this project.

[post]: https://caliburnmicro.com/announcements/stepping-away

**Comment by KasperSK** [comment]
Hi Nigel, 

I am sad to hear that you are stepping down I have really been enjoying your work on Caliburn.Micro, it has long been a favorite framework of mine. 

In your opinion how far away would the version `4.0` that you envisioned be? Are there any major missing feature or is it mostly small issues that need to fixed? I have been using the prerelease version for an upcoming project and for most I think it works and is quite use able.  

Would you be able to act as an consultant for a possible future maintainer? Not on a daily basis but on a once in while sort of deal? 

I would like to thank you for all the work you have put into Caliburn.Micro it has been a joy to use the framework.

**Comment by nigel-sampson** [comment]
I think its reasonably close to finishing, nothing big.  The problem would lie more the between the gap between that version and the documentation.  If it was pushed and someone started a new project with it there isn't much guidance beyond the Features samples.

I'd definitely hang around and help if someone else volunteered. I'd seen some horror stories about frameworks that moved to a new maintainer who immediately pushed malware into the package. What we'd likely start with is commit rights etc while I retain access to the builds, nuget and the domain until such time as I feel comfortable etc.

**Comment by a44281071** [comment]
It's so sad. I have used cal project for 8 years. I have learns a lot from you. 
Also recommended to many Chinese WPF developers.
I will always focus on the project.

**Comment by KasperSK** [comment]
I would love to help out with the documentation and in any other way that I can. I think it is important that we get version `4.0` released it brings a lot of new great features to the table and support for modern platforms. I have colleagues that love using Caliburn.Micro that want to migrato to `netcore` for their future WPF applications and for that i believe they need version `4.0`. 

As for taking over maintaining I am not sure I can make the time commitment, also I am not sure that I am familiar enough with the code base to be able to maintain it. Still I would really love to see Caliburn.Micro live on and evolve.

How is the state of the documentation for `4.0` is it completely missing or has it been started? Could we start by having a migration documentation from `3.2` to `4.0`?

**Comment by nigel-sampson** [comment]
All the documentation is available at https://github.com/Caliburn-Micro/caliburn-micro.github.io

Happy to review PR's against it.

**Comment by nigel-sampson** [comment]
[4.0.136-rc](https://github.com/Caliburn-Micro/Caliburn.Micro/releases/tag/4.0.136-rc) has been published.

**Comment by godrose** [comment]
Sorry to hear that. I’ve been using the framework all across different platforms from 2012. Would love to contribute.

**Comment by paulem** [comment]
I' m so sorry about that.
Unfortunately, I see no decent alternatives. I've been waiting for a stable version that will run on the .NET Core. Hope that you'll be able to complete all the work and make a stable release.

Thank you for your work, Nigel!

**Comment by snalesso** [comment]
Really sad to hear it @nigel-sampson , but I think it was in the air.
I just want you to know that it's years I really love this project, I used it in almost every WPF+MVVM project I started, and I am very grateful to you for the time you spent on it and for how you designed the project: simple but very powerful.
I don't want to waste your time, but since I think you're the most suitable person to ask, take a look at https://github.com/Caliburn-Micro/Caliburn.Micro/issues/709#issuecomment-648869941.

**Comment by KasperSK** [comment]
In the milestone for V4.0 at https://github.com/Caliburn-Micro/Caliburn.Micro/milestones I see 4 issues marked with the bug label should we aim to fix those before releasing? 

They are issues:

1. #618 
2. #517 
3. #521 
4. #621 

Or should they go in the backlog for V4.1?

**Comment by enrico-padovani** [comment]
Hi Nigel,

Thank you for all the efforts in developing and maintaining this project all of these years. I wish you all the best in life! CM is a great piece of code, definitely what a focused framework should be.

I'm currently developing with different technologies but when it comes to WPF I always find myself choosing CM over other solutions both for simplicity and for the convention over configuration approach which @EisenbergEffect a is master of!

I wish the final v4 to see the light, so you can leave the project "in peace" :)

Cheers from Italy!

**Comment by vb2ae** [comment]
I would hate to see this project go away.  I am willing to help maintain it.

**Comment by EisenbergEffect** [comment]
I'd like to extend a huge thanks to @nigel-sampson for all the hard work over the years. When it was time for me to pass the torch, Nigel stepped up and not only accepted the challenge but went above and beyond, taking Caliburn.Micro to the next level. It's no small thing to invest in maintaining an open source library for so many years. Thousands of developers, companies, and hobbyists have prospered thanks to your hard work Nigel. I hope that encourages you and is a source of joy and pride throughout your career. Again, from the depths of my heart, thank you Nigel!

@nigel-sampson It seems there are a few people interested in helping out. How do you see proceeding from here? Though I'm not working in the Xaml space these days, I'm happy to provide general open source guidance to whoever takes the lead in maintaining CM. Let me know if there's any other way I can help out in the transition.

**Comment by nigel-sampson** [comment]
Thanks @EisenbergEffect,

I've seen some horror stories in the Node ecosystem from packages handed over very quickly so I'm happy to take a slower approach where I'm a little more involved. 

If someone is confident I don't mind giving commit rights to the repository while keeping nuget publish gated till trust is established. I can review PR's and make some suggestions, once some history is establish we can expand who has rights to push new packages.

I'm also happy paying the current (very minimal fees) around the domain / dns for the website.

**Comment by CoreyVincent** [comment]
Any more plans? Do we see a future for this project?

**Comment by vb2ae** [comment]
Yes I plan on keeping it going.  I am starting to look at some of the open bugs.

**Comment by KitKat31337** [comment]
@nigel-sampson @EisenbergEffect @vb2ae I have been a long time user of Caliburn.Micro and kicked around in the source at times.  I would be more than happy to take a more active role in making sure it stays around.

**Comment by vb2ae** [comment]
@KitKat31337 thanks.

**Comment by godrose** [comment]
@nigel-sampson @vb2ae @KitKat31337 @EisenbergEffect 
What is the minimal list of issues required to release v4?

**Comment by vb2ae** [comment]
@godrose This is the list. https://github.com/Caliburn-Micro/Caliburn.Micro/milestone/3 personally

**Comment by KasperSK** [comment]
@godrose I think we could release a version now. And I think it is time to do so.

**Comment by vb2ae** [comment]
@KasperSK I agree.

**Comment by KasperSK** [comment]
@vb2ae The only thing the need to be done is to update the Start samples and then I think we are ready to launch 4.0

**Comment by vb2ae** [comment]
@KasperSK Working on the Start Samples.  Pointing to MyGet temporarily until the official NuGet is published.    The PRs in caliburn-micro.github.io would need to completed also.

**Comment by nigel-sampson** [comment]
Sounds good, if you want queue up a PR with an announcement and I'll do them both at the same time.

**Comment by KasperSK** [comment]
@vb2ae do you need any help with the Start Samples?

**Comment by vb2ae** [comment]
@KasperSK  almost done with start samples.  Just have uwp xamarin forms start sample and to test ios ones

**Comment by KasperSK** [comment]
@vb2ae Sounds great, looking forward to the PR.

**Comment by KasperSK** [comment]
@nigel-sampson with the setup projects updated in  #747, is there any thing else you see as outstanding before a release of version 4?

**Comment by nigel-sampson** [comment]
I think we're good,  I'm a bit behind due to some medical things, but assuming the setup PR is good then in a few days I'll push to nuget, create the release on GitHub and publish the blog post.

**Comment by vb2ae** [comment]
@nigel-sampson Thanks Hope you feel better

**Comment by hoangcuongvn** [comment]
I tried several recent MVVM libraries in place of CM. I always ended up with going back to CM. They are not the same level of CM.

**Comment by nigel-sampson** [comment]
I've pushed the new packages to Nuget and published the blog post.

Had some issues with release pipeline so had to do a manual upload of the Xamarin.Forms package.

Would be good to check it out and make sure you're happy with it.

**Comment by KasperSK** [comment]
I Will take a look se if It is all good. Thanks for the the help @nigel-sampson.

**Comment by vb2ae** [comment]
@nigel-sampson thanks.   I will update the startup samples to use the released version,

**Comment by KasperSK** [comment]
@nigel-sampson I think the Xamarin.Forms package is working as intended I did not have any issues upgrading.

**Comment by KasperSK** [comment]
@nigel-sampson Is it about time we get commit right to work directly in the Caliburn.Micro repository?

**Comment by nigel-sampson** [comment]
Yeah, was just thinking that  Will sort something out soon.

**Comment by vb2ae** [comment]
@nigel-sampson that would be nice

**Comment by KasperSK** [comment]
@nigel-sampson would be great :) Looking forward to hear how you think we should move forward.

**Comment by nigel-sampson** [comment]
What I'll probably do this weekend is add you both to the organization as an author as next steps. That's pretty separate from nuget rights and will also let you publish to the website..

**Comment by vb2ae** [comment]
@nigel-sampson thanks
